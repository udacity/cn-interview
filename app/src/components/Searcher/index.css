.searcher {
  display: flex;
  align-items: center;
  flex: 1;
  padding-right: 20px;
  border: 1px solid #dde4e9;
  box-shadow: 2px 2px 8px #dde4e9;
}

.searcher-input {
  display: block;
  padding: 15px;
  font-size: 16px;
  flex: 1;
  color: #9d9d9d;
  border: none;
}

.searcher-input::placeholder {
  color: #9d9d9d;
}

.searcher-input:focus {
  outline: none;
}
