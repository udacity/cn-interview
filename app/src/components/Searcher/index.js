import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MdSearch from 'react-icons/lib/md/search';
import './index.css';

export default class Searcher extends Component {
  static propTypes = {
    onChange: PropTypes.func,
  };

  render() {
    const { onChange } = this.props;
    return (
      <div className="searcher">
        <input
          placeholder="搜索你喜欢的课程"
          type="text"
          className="searcher-input"
          maxLength={50}
          onChange={onChange}
        />
        <MdSearch size={30} color="#1db3e2" />
      </div>
    );
  }
}
