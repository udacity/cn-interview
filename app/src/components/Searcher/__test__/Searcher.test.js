import React from 'react';
import { mount } from 'enzyme';
import Searcher from '..';

it('renders correctly', () => {
  const wrapper = mount(<Searcher />);
  expect(wrapper).toMatchSnapshot();
});

it('can trigger onChange prop', () => {
  const handleChange = jest.fn();
  const wrapper = mount(
    <Searcher onChange={handleChange} />
  );
  wrapper
    .find('input')
    .simulate('change');
  expect(handleChange).toHaveBeenCalled();
});