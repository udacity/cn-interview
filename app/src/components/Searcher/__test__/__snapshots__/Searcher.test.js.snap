// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Searcher>
  <div
    className="searcher"
  >
    <input
      className="searcher-input"
      maxLength={50}
      placeholder="搜索你喜欢的课程"
      type="text"
    />
    <MdSearch
      color="#1db3e2"
      size={30}
    >
      <IconBase
        color="#1db3e2"
        size={30}
        viewBox="0 0 40 40"
      >
        <svg
          fill="currentColor"
          height={30}
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": "#1db3e2",
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width={30}
        >
          <g>
            <path
              d="m15.9 23.4c4.1 0 7.5-3.4 7.5-7.5s-3.4-7.5-7.5-7.5-7.5 3.3-7.5 7.5 3.3 7.5 7.5 7.5z m10 0l8.2 8.2-2.5 2.5-8.2-8.2v-1.4l-0.5-0.4c-1.9 1.6-4.4 2.5-7 2.5-6.1 0-10.9-4.7-10.9-10.7s4.8-10.9 10.9-10.9 10.7 4.8 10.7 10.9c0 2.6-0.9 5.1-2.5 7l0.4 0.5h1.4z"
            />
          </g>
        </svg>
      </IconBase>
    </MdSearch>
  </div>
</Searcher>
`;
